foam_controlwrapper/tests/test_controlwrapper.py

```diff
@@ -6,66 +6,266 @@
+        with self.assertRaises(NotImplementedError):
+            wrapper.add_particles(DataContainer())
+
+    def test_get_particles(self):
+        """Test get_particles method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.get_particles('')
+
+    def test_delete_particles(self):
+        """Test delete_particles method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.delete_particles('')
+
+    def test_iter_particles(self):
+        """Test iter_particles method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.iter_particles()
+
+    def test_add_lattice(self):
+        """Test add_lattice method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.add_lattice('')
+
+    def test_get_lattice(self):
+        """Test get_lattice method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.get_lattice('')
+
+    def test_delete_lattice(self):
+        """Test delete_lattice method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.delete_lattice('')
+
+    def test_iter_lattices(self):
+        """Test iter_lattices method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        with self.assertRaises(NotImplementedError):
+            wrapper.iter_lattices()
+
+    def test_run_time(self):
+        """Test that field variable value is changed after
+        consecutive calls of run method
+
+        """
+
+        wrapper = FoamControlWrapper()
+        name = 'simplemesh'
+        wrapper.CM[CUBA.NAME] = name
+        wrapper.CM_extensions[CUBAExt.GE] = (CUBAExt.INCOMPRESSIBLE,
+                                             CUBAExt.LAMINAR_MODEL)
+        wrapper.SP[CUBA.TIME_STEP] = 1
+        wrapper.SP[CUBA.NUMBER_OF_TIME_STEPS] = 1
+        wrapper.SP[CUBA.DENSITY] = 1.0
+        wrapper.SP[CUBA.DYNAMIC_VISCOSITY] = 1.0
+        wrapper.BC[CUBA.VELOCITY] = {'boundary0': (0.1, 0, 0),
+                                     'boundary1': 'zeroGradient',
+                                     'boundary2': (0, 0, 0),
+                                     'boundary3': 'empty'}
+        wrapper.BC[CUBA.PRESSURE] = {'boundary0': 'zeroGradient',
+                                     'boundary1': 0,
+                                     'boundary2': 'zeroGradient',
+                                     'boundary3': 'empty'}
+        mesh_file = H5CUDS.open(os.path.join('foam_controlwrapper',
+                                             'tests',
+                                             'simplemesh.cuds'))
+        mesh_from_file = mesh_file.get_mesh(name)
```

Comment:
Concerning the NUMBER_OF_TIME_STEPS: Now the solver runs until its convergence criteria is fullfilled or the number of time steps is reached. Should we add also control for the convergence criteria ?
I continued also testing that strange behaviour of consequent run method calls thjat you got and got also peculiar behaviour when i call it in the loop. It seems that the pyFoam's ConvergnceRunner is getting mixed somehow. I wrote simple Runner class and it start working.